src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -31,6 +33,9 @@ export class DatasetService {
     if (this.uiSettings.get(UI_SETTINGS.QUERY_ENHANCEMENTS_ENABLED)) {
       this.registerDefaultTypes();
     }
+    this.recentDatasets = new LRUCache({
```

Comment:
We should store this in session storage so that when the user returns we can show them this. Not a blocker for this PR, but if we are skipping this, lets add a fast follow. Recent datasets are not very useful if they dont persist across my session

Reply:
+1

yeah I agree i think sean and i had similar idea. we should def backlog an item posting.

copy pasta for transparency:

-------
in the interest of timelines, potential approach use hook for setRecentDatasets and then a new const recentOptions to useMemo and then join the the options in the html component

* **create backlog issue**: storing the datasets created in the dataset service
  * new function in dataset service like `getDatasets()`
  * populate the selector for all the datasets. when users create a dataset it stores the dataset service and updates the LRU recent datasets. when user select a dataset it will update the LRU recent datasets. 
  * can use the session storage to store datasets since it data structure could be of type `DATASET`. 
  * when the user opens the dataset selector for the first time it calls getRecentDatasets and and getDatasets from the data service. then subsequent calls will just be the get recent datasets.
* **benefits**: dataset service is a single source of truth of recent datasets and datasets. another plugin can access the datasets without using our component. we can easily update the session id in the dataset service without the query state being updated.